### PR TITLE
Document CRDs and kube namespaces

### PR DESCRIPTION
**Pull request by lkysow**

### CODE REVIEW

File: website/content/docs/agent/config-entries/service-intentions.mdx [hunk 1]
```diff
@@ -108,7 +108,7 @@ spec:
 </Tab>
 </Tabs>
 
-## gRPC
+### gRPC
```
Comment (lkysow) [inline website/content/docs/agent/config-entries/service-intentions.mdx:111]: Fixing wrong-level headings.

File: website/content/docs/k8s/crds/index.mdx [hunk 2]
```diff
@@ -173,3 +173,211 @@ to be deleted since that would put Consul into a broken state.
...
+spec:
+  destination:
+    name: web
+  sources:
+    - name: foo
+      action: allow
+```
+
+~> **NOTE:** If two `ServiceIntentions` resources set the same `spec.destination.name`, the
+one created last will not be synced.
```
Comment (ashwin-venkatesh) [inline website/content/docs/k8s/crds/index.mdx:259]:
```suggestion
last one created will not be synced.
```

File: website/content/docs/k8s/crds/index.mdx [hunk 3]
```diff
@@ -173,3 +173,211 @@ to be deleted since that would put Consul into a broken state.
...
+     name: consul
+     enableConsulNamespaces: true
+     image: hashicorp/consul-enterprise:<tag>-ent
+   connectInject:
+     consulNamespaces:
+       consulDestinationNamespace: 'my-ns'
+   ```
+
+   ~> **NOTE:** In this configuration, if two custom resources of the same kind **and** the same name are attempted to
+   be created in two Kubernetes namespaces, the one created last will not be synced.
```
Comment (ashwin-venkatesh) [inline website/content/docs/k8s/crds/index.mdx:331]:
```suggestion
   be created in two Kubernetes namespaces, the last one created will not be synced.
```

**Review by ashwin-venkatesh — APPROVED** [review]
Looks great!!

File: website/content/docs/k8s/crds/index.mdx [hunk 4]
```diff
@@ -173,3 +173,211 @@ to be deleted since that would put Consul into a broken state.
...
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: web
+  namespace: web-ns
+spec: ...
+```
+
+~> **NOTE:** If two custom resources of the same kind **and** the same name are attempted to
+be created in two Kubernetes namespaces, the one created last will not be synced.
```
[the remark this answers is not in the document]
Comment (lkysow) [reply website/content/docs/k8s/crds/index.mdx:209]: The webhook doesn't reject it, it just fails to reconcile properly and sits as unsynced forever.

**Pull request merged**